src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -134,7 +134,7 @@ public function doAsk(OutputInterface $output, Question $question)
                 $ret = trim($ret);
             }
         } else {
-            $ret = trim($this->autocomplete($output, $question, $inputStream));
+            $ret = trim($this->autocomplete($output, $question, $inputStream, is_array($autocomplete) ? $autocomplete : iterator_to_array($autocomplete, false)));
```

Comment:
passing  `false` to `iterator_to_array` fixes

```
1) Symfony\Component\Console\Tests\Helper\QuestionHelperTest::testTraversableAutocomplete
Undefined offset: 3

src/Symfony/Component/Console/Helper/QuestionHelper.php:294
``` 

in case of assoc array keys (which we dont really care about)